types/views/views.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package views provides read-only accessors for commonly used
+// value types.
+package views
+
+import (
+	"inet.af/netaddr"
+	"tailscale.com/net/tsaddr"
+)
+
+// StringsSlice is a read-only accessor for a slice of strings.
+type StringSlice struct {
+	// ж represents the underlying value.
+	// It is considered a lower-case letter so it is not exported.
```

Comment:
Not worth saying. If anybody tries it they'll learn that soon enough.

I'd say instead:

```go
     // It is named distinctively to make you think of how dangerous it is to escape
     // to callers. You must not let callers be able to mutate it.
```

Reply:
Done.